ui/src/tests/plugin.test.ts

```diff
@@ -0,0 +1,161 @@
+import * as os from 'os';
+import {Page} from 'puppeteer';
+
+import * as auth from './authentication';
+import * as selector from './selector';
+import {GotifyTest, newTest} from './setup';
+import {count, innerText, waitForExists} from './utils';
+
+const pluginSupported = ['linux', 'darwin'].indexOf(os.platform()) !== -1;
+
+let page: Page;
+let gotify: GotifyTest;
+
+beforeAll(async () => {
+    gotify = await newTest(
+        pluginSupported ? ['github.com/gotify/server/plugin/example/echo'] : undefined
+    );
+    page = gotify.page;
+});
+
+afterAll(async () => await gotify.close());
+
+enum Col {
+    ID = 1,
+    SetEnabled = 2,
+    Name = 3,
+    Token = 4,
+    Details = 5,
+}
+
+const hiddenToken = '•••••••••••••••';
+
+const $table = selector.table('#plugin-table');
+
+const switchSelctor = (id: number) => $table.cell(id, Col.SetEnabled, '[data-enabled]');
+
+const enabledState = async (id: number) =>
+    (await page.$eval(switchSelctor(id), (el) => el.getAttribute('data-enabled'))) === 'true';
+
+const toggleEnabled = async (id: number) => {
+    const origEnabled = await enabledState(id).toString();
+    await page.click(switchSelctor(id));
+    await page.waitForFunction(
+        `document.querySelector("${switchSelctor(
+            id
+        )}").getAttribute("data-enabled") !== "${origEnabled}"`
+    );
+};
+
+const pluginInfo = async (className: string) => {
+    return await innerText(page, `.plugin-info .${className} > span`);
+};
+
+const getDisplayer = async () => {
+    return await innerText(page, '.displayer');
+};
+
+const inDetailPage = async (id: number, callback: (() => Promise<void>)) => {
+    await page.click($table.cell(id, Col.Details));
+    await page.waitForFunction(`window.location.toString().endsWith("${id}")`);
```

Comment:
Use waitForExists instead and check for the plugin name: (plugin name should be given via method parameters)
```suggestion
    await waitForExists(page, selector.heading(), 'test plugin')
```